keps/sig-storage/1472-storage-capacity-tracking/README.md

```diff
@@ -774,10 +760,10 @@ enhancement:
   Yes.
 
   When disabling support in the API server, the new object type and
-  thus all objects will disappear together with the new flag in
+  will disappear together with the new flag in
```

Comment:
What exact behavior the feature gate is controlling on the kube-apiserver side?

It seems to me that kube-apiserver is more treated to passing the state and nothing really happens there...


Another question is - how we can control the behavior in the external-provisioner side.

Reply:
> What exact behavior the feature gate is controlling on the kube-apiserver side?

There's no specific logic in kube-apiserver besides offering the API. It has to be a builtin type because there was (and IMHO still is) no way to to ensure that a Kubernetes cluster has a CRD installed. Another unsolved (?) issue is generating Go bindings for a CRD that then can be used by kube-scheduler.

> Another question is - how we can control the behavior in the external-provisioner side.

There are command line flags for it: https://github.com/kubernetes-csi/external-provisioner/blob/master/README.md#capacity-support

Cluster admins can influence that when deploying a CSI driver.

Reply:
OK - I think we need to distringuish two different things in this question:
- when the feature is disabled itself (but no apiserver rollback - so API is still there)
- when there is version rollback (so API is also rolled back)

Can you make those a bit more explicit.
[It's a different thing, but might be work looking into this one: https://github.com/kubernetes/enhancements/pull/2199/files#diff-b18c53698772d9f0dcc15e9f35d90ecd0f181923f33e40115036991aa0b5f1c5R236 ]